ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/context/VariableContext.java

```diff
@@ -80,7 +93,14 @@ public String encodedName() {
     }
 
     public String type() {
-        return type;
+        return this.type;
+    }
+
+    public String typeLHS() {
+        String returnType = this.typeLHS.equals(VAR_TYPE_DESC) &&
+                this.kind.equals(OBJECT) ?
+                VAR : this.type;
```

Comment:
Similar comment to the above one. Plus what's the need to special case objects?

Reply:
Updated in 3fc03c0dee2d7091929456e76b603ab332cb786e

Reply:
As an example in the issue, var type need to assign to the left side of the assignment
var obj = object { function demo() returns int => 42; };  

But when considering function assignments as follows
var add = function (int v1, int v2) returns int { return v1 + v2; };

if we assign var to the left side of the assignment we cant use add variable in later stages
following line raise an error
`>> add(5, 10)`

Because of this reason objects and functions were considered as two different cases.

Reply:
Ah I see. That doesn't sound like a proper solution to the problem. Need to check and figure out why `add()` is not working and then address that. If the user specified the variable using `var`, then it needs to be there in the final generated code as a variable declared using `var`. i.e., remove use of the actual type in LHS for all instances where `var` is used.

Reply:
Removed in d3e7d2d557259712428f4f178b8d6bfd67325435